src/stable_hash_eq.rs

```diff
@@ -0,0 +1,194 @@
+/// Sealed trait for types in [`std`] that are known to implement
```

Comment:
Maybe link to a doc about sealing here? Or explicitly call out that you cannot implement this trait yourself, and that we provide implementations for common types from `std` that we've checked the invariants for.

Reply:
Something like
https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed
?

Reply:
Exactly! And note from that page:
> To avoid frustrated users trying to implement the trait, it should be documented in rustdoc that the trait is sealed and not meant to be implemented outside of the current crate.

Reply:
I suppose since it already says “[Sealed](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) trait for _types in `std`_” there is no potential for confusion anymore?

Reply:
I think it's still better to be explicit. _Many_ users will have no idea what "sealed" means.

Reply:
@jonhoo alright, I’ve added a sentence of explanation